include/irr/video/IGPUVirtualTexture.h

```diff
@@ -0,0 +1,188 @@
+protected:
+    class IGPUVTResidentStorage final : public base_t::IVTResidentStorage
+    {
+        using base_t = base_t::IVTResidentStorage;
+        using cpu_counterpart_t = asset::ICPUVirtualTexture::ICPUVTResidentStorage;
+
+        static core::smart_refctd_ptr<IGPUImage> createGPUImageFromCPU(IVideoDriver* _driver, asset::IAssetManager* _am, asset::ICPUImage* _cpuimg)
+        {
+            auto params = _cpuimg->getCreationParameters();
+            auto img = _driver->createDeviceLocalGPUImageOnDedMem(std::move(params));
+
+            auto regions = _cpuimg->getRegions();
+            if (regions.size())
+            {
+                auto texelBuf = _driver->createFilledDeviceLocalGPUBufferOnDedMem(_cpuimg->getBuffer()->getSize(), _cpuimg->getBuffer()->getPointer());
+                _driver->copyBufferToImage(texelBuf.get(), img.get(), regions.size(), regions.begin());
+            }
+            _am->convertAssetToEmptyCacheHandle(_cpuimg, core::smart_refctd_ptr(img));
+
+            return img;
+        }
+
+    public:
+        IGPUVTResidentStorage(IVideoDriver* _driver, asset::IAssetManager* _am, const cpu_counterpart_t* _cpuStorage) :
+            //TODO awaiting fix: base_t ctor should copy addr alctr state from cpu counterpart
+            base_t(
+                createGPUImageFromCPU(_driver, _am, _cpuStorage->image.get()),
+                _cpuStorage->m_assignedPageTableLayers,
+                _cpuStorage->m_addr_layerShift,
+                _cpuStorage->m_addr_xMask
+            ),
+            m_driver(_driver)
+        {
+
+        }
+
+        IGPUVTResidentStorage(IVideoDriver* _driver, asset::E_FORMAT _format, uint32_t _extent, uint32_t _layers, uint32_t _tilesPerDim) :
+            base_t(_layers, _tilesPerDim),
+            m_driver(_driver)
+        {
+            IGPUImage::SCreationParams params;
+            params.extent = { _extent,_extent,1u };
+            params.format = _format;
+            params.arrayLayers = _layers;
+            params.mipLevels = 1u;
+            params.type = asset::IImage::ET_2D;
+            params.samples = asset::IImage::ESCF_1_BIT;
+            params.flags = static_cast<asset::IImage::E_CREATE_FLAGS>(0);
+
+            image = m_driver->createDeviceLocalGPUImageOnDedMem(std::move(params));
+        }
+
+    private:
+        core::smart_refctd_ptr<IGPUImageView> createView_internal(IGPUImageView::SCreationParams&& _params) const override
+        {
+            return m_driver->createGPUImageView(std::move(_params));
+        }
+
+        IVideoDriver* m_driver = nullptr;
+    };
+
+public:
+    IGPUVirtualTexture(
+        IVideoDriver* _driver,
+        const base_t::IVTResidentStorage::SCreationParams* _residentStorageParams,
+        uint32_t _residentStorageCount,
+        uint32_t _pgTabSzxy_log2 = 8u,
+        uint32_t _pgTabLayers = 256u,
+        uint32_t _pgSzxy_log2 = 7u,
+        uint32_t _tilePadding = 9u,
+        uint32_t _maxAllocatableTexSz_log2 = 14u
```

Comment:
Better defaults, see https://github.com/buildaworldnet/IrrlichtBAW/pull/474/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h&file-filters%5B%5D=No+extension&hide-deleted-files=true#r417504241

Also `_maxAllocatableTexSz_log2` you can work out from `0x1u<<(_pgTabSzxy_log2+_pgSzxy_log2)`